src/java/org/apache/cassandra/metrics/RepairMetrics.java

```diff
@@ -18,17 +18,41 @@
 
 package org.apache.cassandra.metrics;
 
+import java.util.Collections;
+import java.util.EnumMap;
+import java.util.Map;
+
 import com.codahale.metrics.Counter;
+import com.codahale.metrics.Histogram;
+import org.apache.cassandra.net.Verb;
 
 import static org.apache.cassandra.metrics.CassandraMetricsRegistry.Metrics;
 
 public class RepairMetrics
 {
     public static final String TYPE_NAME = "Repair";
     public static final Counter previewFailures = Metrics.counter(DefaultNameFactory.createMetricName(TYPE_NAME, "PreviewFailures", null));
+    private static final Histogram retries = Metrics.histogram(DefaultNameFactory.createMetricName(TYPE_NAME, "Retries", null), false);
+    private static final Map<Verb, Histogram> retriesByVerb = Collections.synchronizedMap(new EnumMap<>(Verb.class));
+    private static final Counter retryTimeout = Metrics.counter(DefaultNameFactory.createMetricName(TYPE_NAME, "RetryTimeout", null));
+    private static final Map<Verb, Counter> retryTimeoutByVerb = Collections.synchronizedMap(new EnumMap<>(Verb.class));
```

Comment:
Are we guaranteed to see metrics in the JMX API if they are registered dynamically after the static context is initialized? I remember using the `DynamicMBean`, but I couldn't find any mention of this in the source code, so I'm asking such a question :-)

Reply:
Fair point... I didn't want to set that up upfront as that requires a list of supported verbs... which can be hard to maintain...

Reply:
ok I moved the test set of verbs that allow retry to `RepairMessage` which enables me to pre-populate